provider/aws/aws.go

```diff
@@ -435,22 +435,26 @@ func (p *AWSProvider) records(ctx context.Context, zones map[string]*profiledZon
 				}
 
 				ep := endpoint.NewEndpointWithTTL(wildcardUnescape(aws.StringValue(r.Name)), aws.StringValue(r.Type), ttl, targets...)
-				if aws.StringValue(r.Type) == endpoint.RecordTypeCNAME {
-					ep = ep.WithProviderSpecific(providerSpecificAlias, "false")
+				if ep != nil {
```

Comment:
This seems like a larger issue across all providers, thoughts on a fix to have NewEndpointWithTTL return an error if we can't create the endpoint and then handling that in all providers rather than failing with returning nil